src/qibojit/tests/conftest.py

```diff
@@ -4,10 +4,9 @@
 qibo.set_backend("qibojit")
 
 _BACKENDS = ["numba"]
-if K._cupy_engine is not None:  # pragma: no cover
+if K.gpu_devices:  # pragma: no cover
     # CI does not test for GPU
-    _BACKENDS.append("cupy")
-
+    _BACKENDS.extend(("cupy", "cuquantum"))
```

Comment:
This should work okay now. 

An additional option that could be useful is to have a command line flag to control which backends are tested, similar to the `--libraries` flag we implemented a few days ago in qibojit-benchmarks. This would be for example if one wants to test numba and cupy without installing cuquantum. Alternatively something like
```
try:
    import cuquantum
    _BACKENDS.append("cuquantum")
except ModuleNotFoundError:
    pass
```
would also work. Let me know your preference.

Reply:
The second option looks simpler. I'll go with that one. If you prefer the other one feel free to change it.